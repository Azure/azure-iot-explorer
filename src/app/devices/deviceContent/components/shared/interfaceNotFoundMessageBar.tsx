/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { MessageBar, MessageBarType } from 'office-ui-fabric-react/lib/MessageBar';
import { MessageBarButton } from 'office-ui-fabric-react/lib/Button';
import { LocalizationContextConsumer, LocalizationContextInterface } from '../../../../shared/contexts/localizationContext';
import { ResourceKeys } from '../../../../../localization/resourceKeys';
import '../../../../css/_interfaceNotFoundMessageBar.scss';

export interface InterfaceNotFoundMessageBoxProps {
    settingsVisibleToggle: (visible: boolean) => void;
}

export default class InterfaceNotFoundMessageBox extends React.Component<InterfaceNotFoundMessageBoxProps> {

    public render() {
        return (
            <LocalizationContextConsumer>
                {(context: LocalizationContextInterface) => (
                    <>
                    <MessageBar
                        messageBarType={MessageBarType.error}
                        actions={
                            <MessageBarButton
                                className="configure-button"
                                onClick={this.handleConfigure}
                            >
                                    {context.t(ResourceKeys.deviceInterfaces.command.configure)}
                            </MessageBarButton>}
                    >
                        {context.t(ResourceKeys.notifications.getInterfaceModelOnError)}
                    </MessageBar>
                    </>
                )}
            </LocalizationContextConsumer>
        );
    }

    private readonly handleConfigure = () => {
        this.props.settingsVisibleToggle(true);
}
}
