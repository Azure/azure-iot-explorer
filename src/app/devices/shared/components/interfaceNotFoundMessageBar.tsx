/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import { MessageBar, MessageBarType, MessageBarButton } from '@fluentui/react';
import { ResourceKeys } from '../../../../localization/resourceKeys';
import { ROUTE_PARTS, ROUTE_PARAMS } from '../../../constants/routes';
import '../../../css/_interfaceNotFoundMessageBar.scss';

export const InterfaceNotFoundMessageBar: React.FC = () => {
    const { t } = useTranslation();
    const history = useHistory();
    const [redirectToModelRepositories, setRedirectToModelRepositories] = React.useState<boolean>(false);

    const onConfigureClick = () => {
        setRedirectToModelRepositories(true);
    };

    if (redirectToModelRepositories) {
        history.push(`/${ROUTE_PARTS.HOME}/${ROUTE_PARTS.MODEL_REPOS}?${ROUTE_PARAMS.NAV_FROM}`);
        return <></>;
    }

    return (
        <div className="message-bar">
            <MessageBar
                messageBarType={MessageBarType.info}
                actions={
                    <MessageBarButton
                        className="configure-button"
                        onClick={onConfigureClick}
                    >
                        {t(ResourceKeys.deviceInterfaces.command.configure)}
                    </MessageBarButton>}
            >
                {t(ResourceKeys.deviceInterfaces.interfaceNotFound)}
            </MessageBar>
        </div>
    );
};
