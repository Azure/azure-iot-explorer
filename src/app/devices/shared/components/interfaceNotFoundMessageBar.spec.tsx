/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import 'jest';
import * as React from 'react';
import { act } from 'react-dom/test-utils';
import { MessageBarButton } from '@fluentui/react';
import { shallow, mount } from 'enzyme';
import { InterfaceNotFoundMessageBar } from './interfaceNotFoundMessageBar';

const histroyPushSpy = jest.fn();

jest.mock('react-router-dom', () => ({
    useHistory: () => ({ push: histroyPushSpy }),
}));

describe('interfaceNotFoundMessageBar', () => {
    it('matches snapshot', () => {
        expect(shallow(<InterfaceNotFoundMessageBar/>)).toMatchSnapshot();
    });

    it ('issues redirect when button clicked', () => {
        const wrapper = mount(<InterfaceNotFoundMessageBar/>);

        act(() => {
            wrapper.find(MessageBarButton).first().props().onClick(undefined);
        });

        wrapper.update();

        expect(histroyPushSpy).toBeCalledWith('/home/repos?from');
    });
});
