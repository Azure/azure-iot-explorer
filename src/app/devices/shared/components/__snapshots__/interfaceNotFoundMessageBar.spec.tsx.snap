// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interfaceNotFoundMessageBar matches snapshot 1`] = `
<div
  className="message-bar"
>
  <StyledMessageBar
    actions={
      <CustomizedMessageBarButton
        className="configure-button"
        onClick={[Function]}
      >
        deviceInterfaces.command.configure
      </CustomizedMessageBarButton>
    }
    messageBarType={0}
  >
    deviceInterfaces.interfaceNotFound
  </StyledMessageBar>
</div>
`;
